### PR TITLE
feat: campaign details header

### PR DESCRIPTION
**Pull request by guerrap**

### CODE REVIEW

File: packages/frontend/src/app/[locale]/campaigns/[chain]/[campaignId]/page.tsx [hunk 1]
```diff
@@ -1,10 +1,11 @@
 import { CampaignDetails } from "@/src/components/campaign-details";
 import type { SupportedChain } from "@metrom-xyz/contracts";
+import type { Address } from "viem";
 
 export default function CampaignDetailsPage({
     params,
 }: {
-    params: { chain: SupportedChain; campaignId: string };
+    params: { chain: SupportedChain; campaignId: Address };
```
Comment (luzzif) [inline packages/frontend/src/app/[locale]/campaigns/[chain]/[campaignId]/page.tsx:8]: This is not right. The campaign id is not an address but 32 bytes. We can use a `Hex` type from `viem`

File: packages/frontend/src/components/campaign-details/index.tsx [hunk 2]
```diff
@@ -1,28 +1,32 @@
...
 import { useRouter } from "@/src/navigation";
 import { Nav } from "../nav";
+import { Header, SkeletonHeader } from "./header";
 
 import styles from "./styles.module.css";
 
 interface CampaignDetailsProps {
     chain: SupportedChain;
-    campaignId: string;
+    campaignId: Address;
```
Comment (luzzif) [inline packages/frontend/src/components/campaign-details/index.tsx:14]: Same as above

File: packages/frontend/src/hooks/useCampaign.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import { type Campaign } from "@metrom-xyz/sdk";
+import { useEffect, useState } from "react";
+import type { Address } from "viem";
+import { metromApiClient, SUPPORTED_AMM_SLUG_TO_NAME } from "../commons";
+
+export interface NamedCampaign extends Campaign {
+    name: string;
+}
+
+export function useCampaign(
```
Comment (luzzif) [inline packages/frontend/src/hooks/useCampaign.ts:11]: Should we accept undefined params here and handle them inside the inner `useEffect`?

File: packages/frontend/src/hooks/useCampaign.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+                if (!cancelled) setLoading(true);
+                const campaign = await metromApiClient.fetchCampaign({
+                    chainId,
+                    id,
+                });
+
+                if (!cancelled)
+                    setCampaign({
+                        ...campaign,
+                        name: `${SUPPORTED_AMM_SLUG_TO_NAME[campaign.pool.amm] || "-"} ${campaign.pool.token0.symbol} / ${campaign.pool.token1.symbol}`,
```
Comment (luzzif) [inline packages/frontend/src/hooks/useCampaign.ts:38]: We could extract this to its own utils function

**Review by luzzif — CHANGES_REQUESTED** [review]

File: packages/frontend/src/app/[locale]/campaigns/[chain]/[campaignId]/page.tsx (hunk 1 above)
Comment (guerrap) [reply packages/frontend/src/app/[locale]/campaigns/[chain]/[campaignId]/page.tsx]: Fixed in 328249fe1c2965f43e5177cfc2abbaf5b371501d.

File: packages/frontend/src/components/campaign-details/index.tsx (hunk 2 above)
Comment (guerrap) [reply packages/frontend/src/components/campaign-details/index.tsx]: Fixed in 328249fe1c2965f43e5177cfc2abbaf5b371501d.

File: packages/frontend/src/hooks/useCampaign.ts (hunk 3 above)
Comment (guerrap) [reply packages/frontend/src/hooks/useCampaign.ts:12]: Fixed in 328249fe1c2965f43e5177cfc2abbaf5b371501d.

File: packages/frontend/src/hooks/useCampaign.ts (hunk 4 above)
Comment (guerrap) [reply packages/frontend/src/hooks/useCampaign.ts]: Fixed in 328249fe1c2965f43e5177cfc2abbaf5b371501d.

**Pull request merged**